core/local/atom/await_write_finish.js

```diff
@@ -107,25 +107,35 @@ function isAggregationCandidate(event) {
   return (
     !event.incomplete &&
     event.kind === 'file' &&
-    ['created', 'modified', 'deleted'].includes(event.action)
+    ['created', 'modified', 'deleted', 'renamed'].includes(event.action)
   )
 }
 
 function aggregateEvents(oldEvent, recentEvent) {
-  if (recentEvent.action === 'deleted' && oldEvent.action === 'created') {
-    // It's just a temporary file that we can ignore
-    log.debug(
-      { createdEvent: oldEvent, deletedEvent: recentEvent },
-      `Ignore ${oldEvent.kind} ${oldEvent.action} then ${recentEvent.action}`
-    )
-
-    return
+  if (recentEvent.action === 'deleted') {
+    if (oldEvent.action === 'created') {
+      // It's just a temporary file that we can ignore
+      log.debug(
+        { createdEvent: oldEvent, deletedEvent: recentEvent },
+        `Ignore ${oldEvent.kind} ${oldEvent.action} then ${recentEvent.action}`
+      )
+
+      return
+    } else if (oldEvent.action === 'renamed') {
+      addDebugInfo(recentEvent, oldEvent)
+
+      recentEvent.path = oldEvent.oldPath
+    }
   }
 
   if (recentEvent.action === 'modified') {
     addDebugInfo(recentEvent, oldEvent)
     // Preserve the action from the first event (it can be a created file)
     recentEvent.action = oldEvent.action
+
+    if (oldEvent.action === 'renamed') {
+      recentEvent.oldPath = oldEvent.oldPath
```

Comment:
I don't understand how this part works: do we have an `oldPath` on `modified` event that is used somewhere?

Reply:
The `recentEvent` action is changed on line 134 into `renamed`. In this case we want to keep the `oldPath` attribute.

Reply:
Does it mean that a `renamed` event can change the checksum of a file?

Reply:
Yes it can. This is the case already when we have to merge the offline edition and movement of a file.